edb/ir/scopetree.py

```diff
@@ -362,8 +402,20 @@ def attach_subtree(self, node: ScopeTreeNode) -> None:
 
         for descendant in node.path_descendants:
             path_id = descendant.path_id.strip_namespace(dns)
-            visible = self.find_visible(path_id)
+            visible, visible_finfo = self.find_visible_ex(path_id)
             if visible is not None:
+                if visible_finfo is not None and visible_finfo.factoring_fence:
+                    # This node is already present in the surrounding
+                    # scope and cannot be factored out, such as
+                    # a reference to a correlated set inside a DML
+                    # statement.
+                    raise InvalidScopeConfiguration(
+                        f'cannot reference correlated set '
+                        f'{path_id.pformat()!r} here',
```

Comment:
not sure I can parse this error message. What does "here" mean in this context?

Reply:
The "here" is pointed to by the parse context arrow.